File: HADObjcExamples/BannerCollectionViewController.m
//
//  BannerCollectionViewController.m
//  HADExamples
//
//  Created by Alexey Fedotov on 27/01/2017.
//  Copyright © 2017 HyperADX. All rights reserved.
//

#import "BannerCollectionViewController.h"
#import <HADFramework/HADFramework.h>

@interface BannerCollectionViewController () <HADBannerAdDelegate>

@property (strong, nonatomic) NSMutableArray *alreadyLoadedAdsInRows;

@end

@implementation BannerCollectionViewController

static NSString * const kDefaultCellIdentifier = @"Cell";
static NSString * const kAdCellIdentifier = @"AD";

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.alreadyLoadedAdsInRows = [NSMutableArray new];
}

#pragma mark <UICollectionViewDataSource>

- (NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView {
    return 1;
}


- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
    return 100;
}

- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
    if (indexPath.row % 9 == 0 && indexPath.row != 0) {
        UICollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:kAdCellIdentifier forIndexPath:indexPath];
        
        if (![self.alreadyLoadedAdsInRows containsObject:indexPath]) {
            [self.alreadyLoadedAdsInRows addObject:indexPath];
            
            HADBannerAd *adView = [[HADBannerAd alloc] initWithPlacementID:@"KoMrp58X" bannerSize:HADBannerAdSizeBanner320x50 viewController:self];
            adView.delegate = self;
            [adView loadAd];
            [cell.contentView addSubview:adView];
            adView.frame = CGRectMake(0, 0, self.view.frame.size.width, 50);
        }
        
        return cell;
    }else{
        return [collectionView dequeueReusableCellWithReuseIdentifier:kDefaultCellIdentifier forIndexPath:indexPath];
    }
}

-(CGSize)collectionView:(UICollectionView *)collectionView layout:(nonnull UICollectionViewLayout *)collectionViewLayout sizeForItemAtIndexPath:(nonnull NSIndexPath *)indexPath{
    if (indexPath.row % 9 == 0 && indexPath.row != 0) {
        return CGSizeMake([UIScreen mainScreen].bounds.size.width, 50);
    } else {
        return CGSizeMake([UIScreen mainScreen].bounds.size.width*0.25, [UIScreen mainScreen].bounds.size.width*0.25);
    }
}

#pragma mark - HADBannerAdDelegate

-(void)hadBannerAdDidLoadWithBannerAd:(HADBannerAd *)bannerAd{
    NSLog(@"hadBannerAdDidLoadWithBannerAd");
}

-(void)hadBannerAdDidClickWithBannerAd:(HADBannerAd *)bannerAd{
    NSLog(@"hadBannerAdDidClickWithBannerAd");
}

-(void)hadBannerAdDidFailWithBannerAd:(HADBannerAd *)bannerAd withError:(NSError *)error{
    NSLog(@"hadBannerAdDidFailWithBannerAd: %@", error);
}

@end

File: HADSwiftExamples/BannerCollectionViewController.swift
//
//  CollectionView.swift
//  HADFramework
//
//  Created by Alexey Fedotov on 09/01/2017.
//  Copyright © 2017 HyperADX. All rights reserved.
//

import Foundation
import HADFramework

class BannerCollectionCell: UICollectionViewCell, HADBannerAdDelegate {
    @IBOutlet weak var bannerContainerView: UIView!
    
    func hadBannerAdDidLoad(bannerAd: HADBannerAd) {
        print("hadViewDidLoad")
    }
    
    func hadBannerAdDidClick(bannerAd: HADBannerAd) {
        print("hadViewDidClick")
    }
    
    func hadBannerAdDidFail(bannerAd: HADBannerAd, withError error: NSError?) {
        print("hadViewDidFail: \(String(describing: error?.localizedDescription))")
    }
}

class BannerCollectionViewController: UICollectionViewController, UICollectionViewDelegateFlowLayout {
    
    var alreadyLoadedAdsInRows: [Int] = []
    
    override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        return 100
    }
    
    override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
        if indexPath.row % 20 == 0 && indexPath.row != 0 {
            
                let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "AD", for: indexPath) as! BannerCollectionCell
            
                if !alreadyLoadedAdsInRows.contains(indexPath.row) {
                    alreadyLoadedAdsInRows.append(indexPath.row)
                    
                    let bannerView = HADBannerAd(placementID: "KoMrp58X", bannerSize:.banner320x50, viewController: self)
                    bannerView.delegate = cell
                    bannerView.loadAd()
                    cell.bannerContainerView?.addSubview(bannerView)
                    
                    //set ad size
                    bannerView.frame = CGRect(x:0, y:0, width:cell.frame.width, height:HADBannerAdSize.getSize(.banner320x50).height)
                }
                return cell
        } else {
            return collectionView.dequeueReusableCell(withReuseIdentifier: "Cell", for: indexPath)
        }
    }
    
    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
        if indexPath.row % 20 == 0 && indexPath.row != 0 {
            return CGSize(width: UIScreen.main.bounds.size.width, height: 50)
        } else {
            return CGSize(width: UIScreen.main.bounds.size.width*0.25, height: UIScreen.main.bounds.size.width*0.25)
        }
    }
}
